util/tls/tls_socket.h

```diff
@@ -116,6 +116,7 @@ class TlsSocket final : public FiberSocketBase {
   enum { WRITE_IN_PROGRESS = 1, READ_IN_PROGRESS = 2, SHUTDOWN_IN_PROGRESS = 4, SHUTDOWN_DONE = 8 };
   uint8_t state_{0};
 
+  // TODO turn this into a class with proper access specifiers
```

Comment:
I will properly turn `AsyncReq` in a class. Will follow up on separate PR once I am done with `AsyncWriteSome`